linux/snap/snapcraft.yaml

```diff
@@ -0,0 +1,62 @@
+name: nextcloud-client
+version: 2.2.4-git
+summary: nextcloud desktop client
+icon: ../client/theme/colored/owncloud-icon.svg
+description: |
+  The ownCloud Desktop Client is a tool to synchronize files from ownCloud
+  Server with your computer.
+
+grade: devel
+confinement: strict
+
+parts:
+  client:
+    plugin: cmake
+    source: ../
+    source-subdir: client
+    build-packages:
+      - libqt5webkit5-dev
+      - libsqlite3-dev
+      - libssl-dev
+      - pkg-config
+      - qt5keychain-dev
+      - qttools5-dev-tools
+    configflags:
+      - -DCMAKE_INSTALL_PREFIX=/usr
+      - -DCMAKE_INSTALL_LIBDIR=/usr/lib
+      - -DDOEM_THEME_DIR=../src/nextcloudtheme
+    after:
+      - desktop-qt5
+      - indicator-qt5
```

Comment:
That is an "hack" that I actually was using (although now you should now use `TMPDIR` set as `XDG_RUNTIME_DIR`, as that is guaranteed to be cleaned up), however when snaps are confined that's not enough because this also would try to connect to dbus to get if running under unity or not.

There's an SRU for appmenu-qt5 in progress which will make this optional, but in the mean time that is the cleanest solution for this problem (see trunk `appmenu-qt5` code).